administration/src/bp-modules/stores/StoresCSVInput.tsx

```diff
@@ -0,0 +1,142 @@
+import { NonIdealState } from '@blueprintjs/core'
+import { parse } from 'csv-parse/browser/esm/sync'
+import React, { ChangeEventHandler, ReactElement, useCallback, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import { StoreFieldConfig } from '../../project-configs/getProjectConfig'
+import { useAppToaster } from '../AppToaster'
+import FileInputStateIcon, { FileInputStateType } from '../FileInputStateIcon'
+import { AcceptingStoreEntry } from './AcceptingStoreEntry'
+import { StoreData } from './StoresImportController'
+import StoresRequirementsText from './StoresRequirementsText'
+
+const StoreImportInputContainer = styled.div`
+  display: flex;
+  align-items: center;
+`
+
+const InputContainer = styled(NonIdealState)`
+  display: flex;
+  align-items: center;
+  flex-grow: 1;
+  justify-content: center;
+`
+type StoresCsvInputProps = {
+  setAcceptingStores: (store: AcceptingStoreEntry[]) => void
+  fields: StoreFieldConfig[]
+}
+
+export const FILE_SIZE_LIMIT_MEGA_BYTES = 2
+const defaultExtensionsByMIMEType = {
+  'text/csv': '.csv',
+}
+const FILE_SIZE_LIMIT_BYTES = FILE_SIZE_LIMIT_MEGA_BYTES * 1000 * 1000
+
+const lineToStoreEntry = (line: string[], headers: string[], fields: StoreFieldConfig[]): AcceptingStoreEntry => {
+  let storeData: StoreData = {}
+  line.forEach((entry, index) => {
+    const columnName = headers[index]
+    // TODO 1570 get geodata if no coordinates available
+    storeData = { ...storeData, [columnName]: entry }
+  })
+  return new AcceptingStoreEntry(storeData, fields)
+}
+
+const StoresCsvInput = ({ setAcceptingStores, fields }: StoresCsvInputProps): ReactElement => {
+  const [inputState, setInputState] = useState<FileInputStateType>('idle')
+  const fileInput = useRef<HTMLInputElement>(null)
+  const appToaster = useAppToaster()
+  const headers = fields.map(field => field.name)
+
+  const showInputError = useCallback(
+    (message: string) => {
+      appToaster?.show({ intent: 'danger', message })
+      setInputState('error')
+      if (!fileInput.current) return
+      fileInput.current.value = ''
+    },
+    [appToaster]
+  )
+
+  const onLoadend = useCallback(
```

Comment:
should it be `onLoadEnd` or `onLoaded` ?

Reply:
yes it is. i will fix that